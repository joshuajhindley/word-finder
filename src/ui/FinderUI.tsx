import React from 'react'
import FinderContainer from '../containers/finderContainer'
import Finder from './Finder'
import HelpModal from './HelpModal'
import ScrollToTop from './ScrollToTop'

const FinderUI = (props: any) => {
  return (
    <FinderContainer {...props}>
      <HelpModal {...props} />
      <Finder {...props} />
      <ScrollToTop {...props} />
    </FinderContainer>
  )
}

export default FinderUI
