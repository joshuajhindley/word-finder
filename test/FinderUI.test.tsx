import FinderUI from '../src/ui/FinderUI'
import createStore from '../src/store'
import { Provider } from 'react-redux'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

const store = createStore()

describe('tests the FinderUI component', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<FinderUI />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Provider store={store}>
        <FinderUI />
      </Provider>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
