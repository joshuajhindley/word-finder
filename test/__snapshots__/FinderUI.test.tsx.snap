// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the FinderUI component matches the snapshot 1`] = `
<Connect(Children)>
  <HelpModal />
  <Finder />
  <ScrollToTop />
</Connect(Children)>
`;
